system/I18n/Time.php

```diff
@@ -81,8 +81,13 @@ public function __construct(?string $time = null, $timezone = null, ?string $loc
 
         // If a test instance has been provided, use it instead.
         if ($time === '' && static::$testNow instanceof self) {
-            $timezone = $timezone ?: static::$testNow->getTimezone();
-            $time     = static::$testNow->format('Y-m-d H:i:s');
+            if ($timezone !== null) {
+                $testNow = static::$testNow->setTimezone($timezone);
```

Comment:
Please clone this; we extend the mutable DateTime.

Reply:
What? Time is immutable, isn't it?